hoomd/CellList.cc

```diff
@@ -84,7 +93,7 @@ uint3 CellList::computeDimensions()
     // size the ghost layer width
     m_ghost_width = make_scalar3(0.0, 0.0, 0.0);
 #ifdef ENABLE_MPI
-    if (m_comm)
+    if (m_sysdef->isDomainDecomposed())
```

Comment:
This is the wrong line. The line I meant is one down.

Reply:
It would probably be safe to lock the weak pointer here as it doesn't have the same deconstruction order issues that using it in the destructor does (see my reply to the nlist comment). However, holding on to the shared pointer guarantees that it will *always* be safe to call `m_comm->method()`. 

It's not about performance, but ensuring that the `Communicator` object accessed by the weak pointer stays alive when it is needed.